From e02e8c3781f54bb86fb4c461256bdc0e9277b85f Mon Sep 17 00:00:00 2001
From: Kra1o5 <kra1o5x@gmail.com>
Date: Mon, 4 Nov 2013 00:39:17 +0100
Subject: [PATCH] Fix-log-spam

---
 services/surfaceflinger/SurfaceFlinger.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index 1a34ee0..2e9fdf2 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -1647,7 +1647,7 @@ void SurfaceFlinger::doComposeSurfaces(const sp<const DisplayDevice>& hw, const
                     case HWC_FRAMEBUFFER_TARGET: {
                         // this should not happen as the iterator shouldn't
                         // let us get there.
-                        ALOGW("HWC_FRAMEBUFFER_TARGET found in hwc list (index=%d)", i);
+                        //ALOGW("HWC_FRAMEBUFFER_TARGET found in hwc list (index=%d)", i);
                         break;
                     }
                 }
-- 
1.8.1.2

